pkg/client/events/replay_client.go

```diff
@@ -0,0 +1,244 @@
+package events
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"cosmossdk.io/depinject"
+
+	"github.com/pokt-network/poktroll/pkg/client"
+	"github.com/pokt-network/poktroll/pkg/either"
+	"github.com/pokt-network/poktroll/pkg/observable"
+	"github.com/pokt-network/poktroll/pkg/observable/channel"
+	"github.com/pokt-network/poktroll/pkg/retry"
+)
+
+const (
+	// eventsBytesRetryDelay is the delay between retry attempts when the events
+	// bytes observable returns an error.
+	eventsBytesRetryDelay = time.Second
+	// eventsBytesRetryLimit is the maximum number of times to attempt to
+	// re-establish the events query bytes subscription when the events bytes
+	// observable returns an error.
+	eventsBytesRetryLimit        = 10
+	eventsBytesRetryResetTimeout = 10 * time.Second
+	// latestObsvblsReplayBufferSize is the replay buffer size of the
+	// latestObsvbls replay observable which is used to cache the latest observable.
+	// It is updated with a new "active" observable when a new
+	// events query subscription is created, for example, after a non-persistent
+	// connection error.
+	latestObsvblsReplayBufferSize = 1
+	// latestReplayBufferSize is the replay buffer size of the latest replay
+	// observable which is notified when block commit events are received
+	// by the events query client subscription created in goPublishEvents.
+	latestReplayBufferSize = 1
```

Comment:
Wdyt about something like this?

```suggestion
	// replayObsCacheBufferSize is the replay buffer size of the
	// replayObsCache replay observable which is used to cache the replay
	// observable that is notified of new events.
	// It, replayObsCache, is updated with a new "active" observable when a new
	// events query subscription is created, for example, after a non-persistent
	// connection error.
	replayObsCacheBufferSize = 1
	// replayObsBufferSize is the replay buffer size of the replay observable
	// which is notified of new events.
	replayObsBufferSize = 1
```